File: src/mx/wawisoft/waaydroid/CardAdapter.java
package mx.wawisoft.waaydroid;

import android.content.Context;
import android.view.Gravity;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.GridView;
import android.widget.TextView;

public class CardAdapter extends BaseAdapter {
    private Context mContext;
    private MagicCardsGenerator magicCardsGenerator;
    private Integer currentCard;
    private Integer cardsNumber;
    
	public CardAdapter(Context c) {
        mContext = c;
        cardsNumber = 5;
        currentCard = 0;
        magicCardsGenerator = new MagicCardsGenerator(cardsNumber);
    }

    public int getCount() {
       return magicCardsGenerator.getCards().get(currentCard).size();
    }

    public Object getItem(int position) {
        return null;
    }

    public long getItemId(int position) {
        return 0;
    }

    // create a new ImageView for each item referenced by the Adapter
    public View getView(int position, View convertView, ViewGroup parent) {
        TextView textView;
        if (convertView == null) {  // if it's not recycled, initialize some attributes
        	textView = new TextView(mContext);
        	textView.setLayoutParams(new GridView.LayoutParams(150, 150));
        	textView.setGravity(Gravity.CENTER);
        	textView.setBackgroundResource(R.drawable.black_border);
        } else {
        	textView = (TextView) convertView;
        }

        textView.setText(magicCardsGenerator.getCards().get(currentCard).get(position).toString());
        return textView;
    }
    
    public void passCard() {
    	currentCard++;
    }
    
    public Integer getCurrentCard() {
		return currentCard;
	}

	public void setCurrentCard(Integer currentCard) {
		this.currentCard = currentCard;
	}
    
	public Integer getCardsNumber() {
		return cardsNumber;
	}
}

File: app/src/main/java/mx/eduardopool/waaydroid/CardAdapter.kt
package mx.eduardopool.waaydroid

import android.content.Context
import android.view.Gravity
import android.view.View
import android.view.ViewGroup
import android.widget.AbsListView
import android.widget.BaseAdapter
import android.widget.TextView

/**

 * Created by EduardoPool on 17/08/14.
 */
class CardAdapter(
        private val mContext: Context,
        var currentCard: Int = 0,
        val cardsNumber: Int = 5,
        private val magicCardsGenerator: MagicCardsGenerator = MagicCardsGenerator(cardsNumber)
) : BaseAdapter() {

    override fun getCount(): Int {
        return magicCardsGenerator.cards[currentCard].size
    }

    override fun getItem(position: Int): Any? {
        return null
    }

    override fun getItemId(position: Int): Long {
        return 0
    }

    // create a new ImageView for each item referenced by the Adapter
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View? {
        val textView: TextView?
        if (convertView == null) {
            // if it's not recycled, initialize some attributes
            textView = TextView(mContext)
            textView.layoutParams = AbsListView.LayoutParams(150, 150)
            textView.gravity = Gravity.CENTER
            textView.setBackgroundResource(R.drawable.black_border)
        } else {
            textView = convertView as TextView?
        }

        textView?.text = magicCardsGenerator.cards[currentCard][position].toString()
        return textView
    }

    fun passCard() {
        currentCard++
    }
}